src/quicr_server_raw_session.cpp

```diff
@@ -274,38 +324,54 @@ QuicRServerRawSession::handle_unsubscribe(
 void
 QuicRServerRawSession::handle_publish(
   const qtransport::TransportContextId& context_id,
-  const qtransport::StreamId& streamId,
+  const qtransport::StreamId& stream_id,
   messages::MessageBuffer&& msg)
 {
   messages::PublishDatagram datagram;
   msg >> datagram;
 
-  auto publish_namespace =
-    std::find_if(publish_namespaces.begin(),
-                 publish_namespaces.end(),
-                 [&datagram](const auto& ns) {
-                   return ns.first.contains(datagram.header.name);
-                 });
-
-  if (publish_namespace == publish_namespaces.end()) {
+  if (publish_state.count(datagram.header.name)) {
     // No such namespace, don't publish yet.
     return;
   }
 
+  auto& [_, states] = *publish_state.find(datagram.header.name);
+
   PublishContext context;
-  if (!publish_state.count(datagram.header.name)) {
+  if (!states.count(context_id)) {
     context.transport_context_id = context_id;
-    context.transport_stream_id = streamId;
+    context.transport_stream_id = stream_id;
+    context.name = datagram.header.name;
 
-    publish_state[datagram.header.name] = context;
+    states[context_id] = context;
   } else {
-    context = publish_state[datagram.header.name];
+    context = states[context_id];
   }
 
-  delegate.onPublisherObject(context.transport_context_id,
-                             context.transport_stream_id,
-                             false,
-                             std::move(datagram));
+  // Check if this is the datagram stream, and react accordingly if not.
+  bool new_stream = false;
+  if (stream_id >= 4) {
```

Comment:
We have been leveraging that stream ID must start at 1 for a sequence number. This resulted in the `stream_id` always being `>= 4`.   It does simplify code a bit.... but RFC9000 only states that 0  - `2**62-1` are valid, leaving 0, 1, 2, 3 as valid stream IDs.   We have been using stream ID 0 and 1 to indicate datagram.  Picoquic has different callbacks, so it's possible we add to the context if the message is datagram or stream based. This would allow us to support stream_ids being 0, 1, 2, 3.

Reply:
Aha! I knew I left something weird in here somewhere. Good catch!

Reply:
This is still in the code right now, guidance on a better way for this is appreciated.